amf-webapi/shared/src/main/scala/amf/plugins/domain/shapes/validation/ShapesNodesValidator.scala

```diff
@@ -30,14 +31,20 @@ object ShapesNodesValidator {
       .groupBy(_.shape)
       .keys
       .flatMap({
-        case s: Shape if s.values.nonEmpty =>
-          s.values.map(v => ValidationCandidate(s, PayloadFragment(v, defaultMediaTypeFor(v))))
-        case _ => Nil
+        case s: Shape if s.values.nonEmpty => shapeEnumCandidates(s)
+        case _                             => Nil
       })
     //call to validation
     PayloadValidationPluginsHandler.validateAll(enumsCandidates.toSeq, severity, env)
   }
 
+  def shapeEnumCandidates(shape: Shape): Seq[ValidationCandidate] = {
```

Comment:
Make private. Should we add something that tests this? I fear someone will edit it and won't realize it is a mistake until a critical issue arrives.